src/link/filelink.cpp

```diff
@@ -56,9 +67,17 @@ bool FileLink::setConfiguration(const LinkConfiguration& linkConfiguration)
 }
 
 bool FileLink::startConnection() {
+    // WriteOnly is used only to save data
+    if(_openModeFlag == QIODevice::WriteOnly) {
+        // The file will be created when something is received
+        // Avoiding empty files
+        // Check if path is writable
+        return QFileInfo(QFileInfo(_file).canonicalPath()).isWritable();
+    }
+
+    // Everything after this point is to deal with reading data
     bool ok = _file.open(QIODevice::ReadWrite);
-    // FileLink created as read ?
-    if(ok && _openModeFlag == QIODevice::ReadOnly) {
+    if(ok) {
         Pack pack;
         if(_logThread) {
```

Comment:
I take it everything here is to do with replaying logs, do you think this should be factored out of this class at some point in the future?

Reply:
Maybe only to create the thread and struct itself, but right now this code is only 30 lines. If it gets more complex in the future, we can create a class to deal with it.